### PR TITLE
config: define workflow scheduling policies

### PR DESCRIPTION
**Pull request by audrium**
closes https://github.com/reanahub/reana-server/issues/357

### CODE REVIEW

File: reana_commons/config.py [hunk 1]
```diff
@@ -226,6 +226,12 @@ def kubernetes_node_label_to_dict(node_label):
 }
 """REANA Workflow Controller address."""
 
+REANA_WORKFLOW_SCHEDULING_POLICIES = ["fifo", "balanced"]
+"""REANA workflow scheduling policies.
+- ``fifo``: first-in-first-out strategy regardless of users.
+- ``balanced``: multi-user-aware strategy.
```
Comment (tiborsimko) [inline reana_commons/config.py:232]: balanced: a weighted strategy taking into account existing multi-user workloads and the DAG complexity of incoming workflows


... would be more precise, albeit a bit longer.

File: reana_commons/config.py [hunk 2]
```diff
@@ -226,6 +226,12 @@ def kubernetes_node_label_to_dict(node_label):
 }
 """REANA Workflow Controller address."""
 
+REANA_WORKFLOW_SCHEDULING_POLICIES = ["fifo", "balanced"]
+"""REANA workflow scheduling policies.
+- ``fifo``: first-in-first-out strategy regardless of users.
```
Comment (tiborsimko) [inline reana_commons/config.py:231]: first-in first-out strategy starting workflows as they come

**Review by mvidalgarcia — COMMENTED** [review]
Any reason why we need this config variable in `reana-commons`? Maybe it can just live in `reana-server`?

**Comment by audrium** [comment]
> Any reason why we need this config variable in `reana-commons`? Maybe it can just live in `reana-server`?

I guess the only reason is that it could be reused in the future if needed in other components. But you are right, since now it's only needed in `reana-server`, it could be moved there

**Comment by mvidalgarcia** [comment]
> I guess the only reason is that it could be reused in the future if needed in other components. But you are right, since now it's only needed in `reana-server`, it could be moved there

It might happen, but in principle, the scheduler lives in `reana-server` component so it's unlikely that it happens, not 100% sure though.

**Comment by audrium** [comment]
> > I guess the only reason is that it could be reused in the future if needed in other components. But you are right, since now it's only needed in `reana-server`, it could be moved there
> 
> It might happen, but in principle, the scheduler lives in `reana-server` component so it's unlikely that it happens, not 100% sure though.

Moved `REANA_WORKFLOW_SCHEDULING_POLICIES` to `reana-server` configs. Closing

**Pull request closed without merging**